js/pivottable/export_renderers.js

```diff
@@ -67,14 +75,73 @@
           r = result[n];
           text += r.join("\t") + "\n";
         }
-        return $("<textarea>").text(text).css({
-          width: ($(window).width() / 2) + "px",
-          height: ($(window).height() / 2) + "px"
+
+        return $('<a id="download" href="data:text/tsv,' + encodeURIComponent(text) + '"> Download as a TSV File </a>').click(function() {
+          $('#download').attr('download', 'pivot_report_' + getcurrenttimestamp() + '.tsv');
+        });
+     },
+
+    "CSV Export": function(pivotData, opts) {
+        var agg, colAttrs, colKey, colKeys, defaults, i, j, k, l, len, len1, len2, len3, len4, len5, m, n, r, result, row, rowAttr, rowAttrs, rowKey, rowKeys, text;
```

Comment:
Can we use any 3rd party library to achieve this? To be honest, current code is not readable at all
so if we cant get a 3rd party library(preferred option), I would suggest,
Break the function into smaller sub functions

```
"CSV Export": function(pivotData, opts) {
 doStep1(); //provide proper names like, handleRows, handleColumns etc 
 ....
 doStepN();
}
```
Also add code comments in each function explaining what is happening